custom/yeksi_naa_reports/sqldata.py

```diff
@@ -809,6 +809,87 @@ def headers(self):
         return headers
 
 
+class SatisfactionRateAfterDeliveryData(VisiteDeLOperateurPerProductDataSource):
+    slug = 'satisfaction_rate_after_delivery'
+    comment = '% products ordered vs. delivered'
+    title = 'Satisfaction Rate after delivery'
+    show_total = True
+    custom_total_calculate = True
+
+    def calculate_total_row(self, products):
+        total_row = ['Total (CFA)']
+        for i in range(len(self.months)):
+            total_row.append(self.percent_fn(
+                sum(
+                    products[product_id][i]['numerator'] for product_id in products if
+                    products[product_id][i]['numerator'] and products[product_id][i]['denominator']
```

Comment:
maybe not important, but i'm curious why a numerator is required here? 

wouldn't 0/100 be a relevant data point to include in a total percentage?

Reply:
Right, it should only check denominator existence to include only relevant data.